src/mlpack/core/util/using.hpp

```diff
@@ -101,6 +92,20 @@ struct GetFillType
   static constexpr const decltype(coot::fill::randu)& randu = coot::fill::randu;
   static constexpr const decltype(coot::fill::randn)& randn = coot::fill::randn;
 };
+
+#else
+
+// By default, assume that we are using an Armadillo object.
+template<typename MatType>
+struct GetFillType
+{
+  static constexpr const decltype(arma::fill::none)& none   = arma::fill::none;
+  static constexpr const decltype(arma::fill::zeros)& zeros = arma::fill::zeros;
+  static constexpr const decltype(arma::fill::ones)& ones   = arma::fill::ones;
+  static constexpr const decltype(arma::fill::randu)& randu = arma::fill::randu;
+  static constexpr const decltype(arma::fill::randn)& randn = arma::fill::randn;
+};
+
 #endif
```

Comment:
Up to you, but you could integrate the `GetFillType` definitions into the `#ifdef` blocks above.  Either way is fine.

Reply:
I would keep it separated for now, easier to read.